tests/unittests/tests-crypto/tests-crypto-sha256.c

```diff
@@ -17,48 +18,75 @@
 
 #include "tests-crypto.h"
 
+static int compare_str_vs_digest(const char *str,
+                                 const unsigned char hash[SHA256_DIGEST_LENGTH])
+{
+    char ch[3] = { 0, 0, 0 };
+    size_t iter_hash = 0;
+
+    for (size_t i = 0; i < strlen(str); i += 2) {
+        ch[0] = str[i];
+        ch[1] = str[i + 1];
+        char *end;
+
+        if (hash[iter_hash++] != strtol(ch, &end, 16)) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
 static int calc_and_compare_hash(const char *str, const char *expected)
 {
     static unsigned char hash[SHA256_DIGEST_LENGTH];
     sha256_context_t sha256;
+
```

Comment:
the spaces and empty lines below are a result of uncrustification, so basically unrelated.